DigitalLearningSolutions.Data/Services/SectionContentService.cs

```diff
@@ -0,0 +1,77 @@
+    {
+        SectionContent? GetSectionContent(int customisationId, int candidateId, int sectionId);
+    }
+
+    public class SectionContentService : ISectionContentService
+    {
+        private readonly IDbConnection connection;
+        private readonly ILogger<SectionContentService> logger;
+
+        public SectionContentService(IDbConnection connection, ILogger<SectionContentService> logger)
+        {
+            this.connection = connection;
+            this.logger = logger;
+        }
+
+        public SectionContent? GetSectionContent(int customisationId, int candidateId, int sectionId)
+        {
+            return connection.QueryFirstOrDefault<SectionContent>(
+                @"
+                    SELECT
+                        Customisations.CustomisationName,
+                        Applications.ApplicationName,
+                        Sections.SectionName,
+                        SUM(aspProgress.TutTime) AS TimeMins, 
+                        Sections.AverageSectionMins AS AverageSectionTime, 
+                        dbo.CheckCustomisationSectionHasLearning(Customisations.CustomisationID, Sections.SectionID) AS HasLearning,
+                        (CASE
+                            WHEN Progress.CandidateID IS NULL
+                                OR dbo.CheckCustomisationSectionHasLearning(Progress.CustomisationID, Sections.SectionID) = 0
+                            THEN 0
+                            ELSE CAST(SUM(aspProgress.TutStat) * 100 AS FLOAT) / (COUNT(Tutorials.TutorialID) * 2)
+                        END) AS PercentComplete
+                    FROM
+                        Tutorials
+                        INNER JOIN CustomisationTutorials
+                            ON CustomisationTutorials.TutorialID = Tutorials.TutorialID
+                        INNER JOIN Customisations
+                            ON Customisations.CustomisationID = CustomisationTutorials.CustomisationID
+                        INNER JOIN Applications
+                            ON Applications.ApplicationID = Customisations.ApplicationID
+                        INNER JOIN Sections
+                            ON Sections.SectionID = Tutorials.SectionID
+                        LEFT JOIN Progress
+                            ON Progress.CustomisationID = Customisations.CustomisationID
+                            AND Progress.CandidateID = @candidateId
+                        LEFT JOIN aspProgress
+                            ON aspProgress.TutorialID = CustomisationTutorials.TutorialID
+                            and aspProgress.ProgressID = Progress.ProgressID
+                    WHERE
+                        CustomisationTutorials.CustomisationID = @customisationId
+	                    AND Sections.SectionID = @sectionId
+	                    AND (Sections.ArchivedDate IS NULL)
+	                    AND (CustomisationTutorials.DiagStatus = 1 OR Customisations.IsAssessed = 1 OR CustomisationTutorials.Status = 1)
```

Comment:
I think we should have some more tests checking that this `WHERE` is working correctly, i.e. having a test checking that:
* an archived section is not returned
* a section where DiagStatus != 1 && IsAssessed != 1 && Status != 0 is not returned
* a section when only one of `CustomisationTutorials.DiagStatus = 1 OR Customisations.IsAssessed = 1 OR CustomisationTutorials.Status = 1` is true is returned

Reply:
Yep I'll add those in :D 
Should it be Status != 1?

Reply:
Yes, sorry 🤦